components/imageproc/src/helpers.rs

```diff
@@ -32,17 +35,16 @@ pub fn fix_orientation(img: &DynamicImage, path: &Path) -> Option<DynamicImage>
     }
 }
 
-/// We only use the input_path to get the file stem.
-/// Hashing the resolved `input_path` would include the absolute path to the image
-/// with all filesystem components.
 pub fn get_processed_filename(
     input_path: &Path,
-    input_src: &str,
     op: &ResizeOperation,
     format: &Format,
-) -> String {
+) -> Result<String> {
     let mut hasher = DefaultHasher::new();
-    hasher.write(input_src.as_ref());
+    hasher.write(
```

Comment:
does that change perf significantly? I can imagine reading a whole site of images is going to be much more computationally expensive than filenames

Reply:
Also it doesn't make sense to read the file multiple times, once for the filename and once to actually operate on it

Reply:
Yeah, I've been thinking about how to approach this. We'd essentially want to read each input once (at most), and for each operation to hold a reference or key to the in-memory contents of the input for processing.